### PR TITLE
parser: remove the old lexer, clean up

### PR DESCRIPTION
**Pull request by tiancaiamao**

### CODE REVIEW

File: parser/parser_test.go [hunk 1]
```diff
@@ -1024,50 +1023,3 @@ func BenchmarkParse(b *testing.B) {
...
-	`show tables like 'show\_test'`,
-	`select _utf8"string";`,
-	"alter table show_test drop foreign key `fk`",
-	`insert t values('\x01')`,
-	`select "ab\_c"`,
-	`select "ab\%c"`,
-	`SELECT CONVERT("ABCD" USING ASCII);`,
-}
-
-func (s *testParserSuite) TestParserCompatible(c *C) {
```
Comment (shenli) [inline parser/parser_test.go]: Should we port this test case to other place?

**Comment by shenli** [comment]
@coocood PTAL

File: parser/parser_test.go (hunk 1 above)
Comment (tiancaiamao) [inline parser/parser_test.go]: Not necessary.

The main purpose of those test cases is for easy debugging. when I found some statement can't work, I add them here, then fix the problem.

So all of those tests should be found some where else, maybe from tidb-test or other place.

**Comment by shenli** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**